pkg/cloudencrypt/legacy.go

```diff
@@ -0,0 +1,44 @@
+package cloudencrypt
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+)
+
+// LegacyEncryptor encrypts values using the main encryptor but is able to decode
+// using other encryptors as well based on the prefix of the given encrypted value.
+type LegacyEncryptor struct {
+	encryptor  *PrefixEncryptor
+	encryptors []*PrefixEncryptor
```

Comment:
```suggestion
	decryptors []*PrefixEncryptor
```
We use these only for `Decrypt` and `Close` methods. We should never do `Encrypt` with them in this `LegacyEncryptor` case